src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -0,0 +1,270 @@
+  Group & m_group;
+
+};
+
+/// Dedicated implementation of GroupContext for Wrapper.
+/// See DataContext class for more info.
+class WrapperContext final : public GroupContext
+{
+public:
+
+  /**
+   * @brief Construct a new WrapperContext object
+   * @param wrapper the target Wrapper object
+   */
+  WrapperContext( WrapperBase & wrapper );
+
+  /**
+   * @brief Destroy the WrapperContext object
+   */
+  virtual ~WrapperContext() {}
+
+  /**
+   * @return the parent group DataContext followed by the wrapper name.
+   */
+  virtual string toString() const;
+
+};
+
+/// Stores information to retrieve where a Group or Wrapper has been declared in the input source file (e.g. XML)
+class DataFileContext final : public DataContext
+{
+public:
+
+  /**
+   * @brief Construct the file context of a Group from an xml node.
+   * @param group the target Group object
+   * @param nodePos the target object xml node position
+   * @param nodeTagName the xml node tag name.
+   */
+  DataFileContext( Group & group, xmlWrapper::xmlNodePos const & nodePos, string const & nodeTagName );
+  /**
+   * @brief Construct the file context of a Group from an xml attribute.
+   * @param wrapper the target Wrapper object
+   * @param attPos the target object xml attribute position
+   */
+  DataFileContext( WrapperBase & wrapper, xmlWrapper::xmlAttributePos const & attPos );
+
+  /**
+   * @brief Destroy the DataFileContext object
+   */
+  virtual ~DataFileContext() {}
+
+  /**
+   * @return the target object name followed by the the file and line declaring it.
+   */
+  virtual string toString() const;
+
+  /**
+   * @return the type name in the source file (XML node tag name / attribute name).
+   */
+  string getTypeName() const
+  { return m_typeName; }
+
+  /**
+   * @return the source file path where the target object has been declared.
+   */
+  string getFilePath() const
+  { return m_filePath; }
+
+  /**
+   * @return the line (starting from 1) where the target object has been declared in the source file.
+   */
+  size_t getLine() const
+  { return m_line; }
+
+  /**
+   * @return the character offset in the line (starting from 1) where the target object has been
+   * declared in the source file.
+   */
+  size_t getOffsetInLine() const
+  { return m_offsetInLine; }
+
+  /**
+   * @return the character offset (starting from 0) in the source file path where the target object has been
+   * declared.
+   */
+  size_t getOffset() const
+  { return m_offset; }
+
+protected:
+
+  /// see getTypeName()
+  string const m_typeName;
+  /// see getFilePath()
+  string const m_filePath;
+  /// see getLine()
+  size_t const m_line;
+  /// see getLineOffset()
+  size_t const m_offsetInLine;
+  /// see getOffset()
+  size_t const m_offset;
+
+};
+
+
+} /* namespace dataRepository */
+
+} /* namespace geos */
+
+
+
+/// @cond DO_NOT_DOCUMENT
+#ifdef GEOSX_USE_FMT
+#define GEOS_FMT_NS_PREFIX fmt
+#else
+#define GEOS_FMT_NS_PREFIX std
```

Comment:
Aren't those already defined somewhere else?

Reply:
Oh ! `GEOS_FMT_NS` changed from `::fmt` to `fmt` (and `std` define too).

The old form forced me to add these defines, but they are no longer relevant.